src/components/webViewRe.re

```diff
@@ -0,0 +1,289 @@
+module type WebViewComponent = {
+  type webViewNavState;
+  let webViewNavState:
+    url::string? =>
+    title::string? =>
+    loading::bool? =>
+    canGoBack::bool? =>
+    canGoForward::bool? =>
+    unit =>
+    webViewNavState;
+  type webViewIOSLoadRequestEvent;
+  let webViewIOSLoadRequestEvent:
+    target::int =>
+    canGoBack::bool =>
+    lockIdentifier::int =>
+    loading::bool =>
+    title::string =>
+    canGoForward::bool =>
+    navigationType::[ | `other | `click] =>
+    url::string =>
+    unit =>
+    webViewIOSLoadRequestEvent;
+  type webViewSource;
+  let webViewSource:
+    uri::string? => method::string? => headers::Js.t 'a? => body::string? => unit => webViewSource;
+  type webViewContentInsets;
+  let webViewContentInsets:
+    top::int? => left::int? => bottom::int? => right::int? => unit => webViewContentInsets;
+  let make:
+    accessibilityLabel::ReasonReact.reactElement? =>
+    accessible::bool? =>
+    hitSlop::TypesRN.insets? =>
+    onAccessibilityTap::(unit => unit)? =>
+    onLayout::(RNEvent.NativeLayoutEvent.t => unit)? =>
+    onMagicTap::(unit => unit)? =>
+    responderHandlers::Props.touchResponderHandlers? =>
+    pointerEvents::[ | `auto | `none | `boxNone | `boxOnly]? =>
+    removeClippedSubviews::bool? =>
+    style::StyleRe.t? =>
+    testID::string? =>
+    accessibilityComponentType::
+      [ | `none | `button | `radiobutton_checked | `radiobutton_unchecked]? =>
+    accessibilityLiveRegion::[ | `none | `polite | `assertive]? =>
+    collapsable::bool? =>
+    importantForAccessibility::[ | `auto | `yes | `no | `noHideDescendants]? =>
+    needsOffscreenAlphaCompositing::bool? =>
+    renderToHardwareTextureAndroid::bool? =>
+    accessibilityTraits::
+      list [
+        | `none
+        | `button
+        | `link
+        | `header
+        | `search
+        | `image
+        | `selected
+        | `plays
+        | `key
+        | `text
+        | `summary
+        | `disabled
+        | `frequentUpdates
+        | `startsMedia
+        | `adjustable
+        | `allowsDirectInteraction
+        | `pageTurn
+      ]? =>
+    accessibilityViewIsModal::bool? =>
+    shouldRasterizeIOS::bool? =>
+    automaticallyAdjustContentInsets::bool? =>
+    contentInsets::webViewContentInsets? =>
+    injectJavaScript::(string => unit)? =>
+    injectedJavaScript::string? =>
+    mediaPlaybackRequiresUserAction::bool? =>
+    onError::(webViewNavState => unit)? =>
```

Comment:
At the moment is `webViewNavState` an abstract type, so you cannot do anything with it.
What you actually want is that onError has the signature
```reason
Js.t {. target: int, canGoBack: Js.boolean, lockIdentifier: int, ...} => unit
```